src/events/pollVotationResults.js

```diff
@@ -8,51 +8,62 @@ module.exports = {
   async execute(message) {
     const { embeds, author, client, channelId } = message;
 
-    if (
-      !author.bot ||
-      !Array.isArray(embeds[0]?.fields) ||
-      embeds[0]?.data?.type !== 'poll_result'
-    ) {
+    if (!author.bot || !embeds || embeds[0]?.data?.type !== 'poll_result') {
       return;
     }
 
-    const userMentionField = embeds[0]?.fields.find(
-      (field) => field.name === 'poll_question_text'
-    );
+    try {
+      const questionField = embeds[0]?.fields.find(
+        (field) => field.name === 'poll_question_text'
+      );
 
-    if (userMentionField === undefined) {
-      return;
-    }
+      if (!questionField) {
+        return;
+      }
 
-    const parts = userMentionField.value?.split('|').map((part) => part.trim());
-    if (!Array.isArray(parts) || !parts[1]) {
-      return;
-    }
+      const parts = questionField.value.split('|').map((part) => part.trim());
+      if (!parts || parts.length < 2) {
+        return;
+      }
+
+      const userMentioned = `<@${parts[1]}>`;
+      const pointType = parts[0].toLowerCase().includes('boosted')
+        ? 'boosted'
+        : 'normal';
 
-    const userMentioned = `<@${parts[1]}>`;
+      const selectedTagId =
+        pointType === 'boosted'
+          ? tagIds.boostedPointTagId
+          : tagIds.addPointTagId;
 
-    const channel = await client.channels.fetch(channelId);
+      const finalResultField = embeds[0]?.fields.find(
+        (field) => field.name === 'victor_answer_text'
+      );
+      const finalResult = parseInt(finalResultField?.value || '0', 10);
 
-    if (channel) {
-      if (embeds[0].fields.length > 3) {
-        const finalResult = embeds[0].fields.find(
-          (result) => result.name === 'victor_answer_text'
-        ).value;
+      const channel = await client.channels.fetch(channelId);
 
-        await Promise.all(
-          Array.from({ length: finalResult }).map(async () => {
-            await channel.send(
-              `<@&${tagIds.boostedPointTagId}> ${userMentioned}`
-            );
-          })
-        );
+      if (channel) {
+        if (finalResult > 0) {
+          await Promise.all(
+            Array.from({ length: finalResult }).map(async () => {
+              await channel.send(`<@&${selectedTagId}> ${userMentioned}`);
+            })
+          );
+        } else {
+          await channel.send(
+            `The result was not valid or no points were awarded.`
+          );
+        }
       } else {
-        await channel.send(`The draw is not suported`);
+        await message.reply({
+          content: `Could not find the channel: ${channelId}`,
+          ephemeral: true,
+        });
```

Comment:
Maybe early return is better to avoid nested ifs, something like:
```js
     if (!channel)  {
       return await message.reply({
            content: `Could not find the channel: ${channelId}`,
            ephemeral: true,
          });
    }

   if (!finalResult || isNaN(finalResult)) {
       return await channel.send(
        `The result was not valid or no points were awarded.`
      );
   }

    await Promise.all(
      Array.from({ length: finalResult }).map(async () => {
        await channel.send(`<@&${selectedTagId}> ${userMentioned}`);
      })
    );
```